### PR TITLE
Refactoring tours

### PR DESCRIPTION
**Pull request by VNTee**
Refactoring 3 methods in service and controller layer:
-addTour
-updateTour
-searchTour

Resolve BUG with SQL exception on the front-end

Delete my own "validator"

### CODE REVIEW

File: src/main/java/com/epam/controller/ToursController.java [hunk 1]
```diff
@@ -66,15 +69,24 @@ public ModelAndView searchTours(@RequestParam String country, @RequestParam Stri
...
         try {
-            LocalDate addStartDate = Validator.getDate(startDate, true);
-            LocalDate addEndDate = Validator.getDate(endDate, true);
-            toursModel.addObject("hotels", hotelService.getMapOfHotels());
-            toursModel.addObject("list", toursOfferService.searchTours(country, addStartDate, addEndDate));
-            return toursModel;
-        } catch (Exception e) {
-            toursModel.addObject("error", e.getMessage());
-            return toursModel;
+            if(!startDate.isEmpty()) {
```
Comment (goponenkoev) [inline src/main/java/com/epam/controller/ToursController.java]: I think u can collapse it. For example, don't allocate new vars. Inline in method's parameters `Date.valueOf() `. If statements here unnecessary, 'cause u have `try-catch`. Just put your code after `catch `inside that block and add `IllegalArgsExeption`.

File: src/main/java/com/epam/controller/ToursController.java [hunk 2]
```diff
@@ -66,15 +69,24 @@ public ModelAndView searchTours(@RequestParam String country, @RequestParam Stri
                                     @RequestParam String endDate, @RequestParam String numberOfPeople) {
```
Comment (goponenkoev) [inline src/main/java/com/epam/controller/ToursController.java:67]: Do u use this arg `numberOfPeople `somewhere?

File: src/main/java/com/epam/controller/ToursController.java [hunk 3]
```diff
@@ -102,64 +114,83 @@ public ModelAndView addTour(@RequestParam String tourType, @RequestParam String
...
         try {
-            LocalDate addStartDate = Validator.getDate(startDate, false);
-            LocalDate addEndDate = Validator.getDate(endDate, false);
-            Integer addPricePerPerson = Validator.getInt(pricePerPerson);
-            Integer addDiscount = Validator.getDiscount(discount);
-            Validator.checkEmpty(tourType);
-            Validator.checkEmpty(tourDescription);
-            Validator.dateDifference(addStartDate,addEndDate);
+            if(!startDate.isEmpty()) {
+                addStartDate = Date.valueOf(startDate).toLocalDate();
```
Comment (goponenkoev) [inline src/main/java/com/epam/controller/ToursController.java]: Please think about collapsing this code. I see several places with duplicate logic

**Review by goponenkoev — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/controller/ToursController.java (hunk 2 above)
Comment (VNTee) [reply src/main/java/com/epam/controller/ToursController.java:67]: No, its unnecessary

File: src/main/java/com/epam/controller/ToursController.java (hunk 1 above)
Comment (VNTee) [reply src/main/java/com/epam/controller/ToursController.java]: Done

File: src/main/java/com/epam/controller/ToursController.java (hunk 3 above)
Comment (VNTee) [reply src/main/java/com/epam/controller/ToursController.java]: Done

File: src/main/java/com/epam/validator/Validator.java [hunk 4]
```diff
@@ -1,93 +1,60 @@
...
+        }
+    }
 
-   @SneakyThrows
-    public static void checkEmpty(String inputString){
-       String nullPointerString = "String is empty. User current input String: ";
-        if(inputString.isEmpty()){
-            log.error(nullPointerString+inputString);
-            throw new IllegalArgumentException(nullPointerString+inputString);
+    public static String getDescriptionString(String description) {
```
Comment (ghost) [inline src/main/java/com/epam/validator/Validator.java:46]: excellent logic))))

File: src/main/java/com/epam/validator/Validator.java [hunk 5]
```diff
@@ -1,93 +1,60 @@
...
 
-   @SneakyThrows
-    public static void dateDifference(LocalDate firstDate, LocalDate secondDate){
-       String differenceIssue = "Negative or zero difference between dates. User current input Dates difference: ";
-       if(!(DAYS.between(firstDate,secondDate)>=1)) {
-           log.error(differenceIssue + DAYS.between(firstDate,secondDate));
-           throw new IllegalArgumentException(differenceIssue + DAYS.between(firstDate,secondDate));
-       }
-   }
+    public static String getTourTypeString(String tourType) {
```
Comment (ghost) [inline src/main/java/com/epam/validator/Validator.java:54]: please, try to use library methods

**Review by ghost — CHANGES_REQUESTED** [review]

File: src/main/java/com/epam/validator/Validator.java (hunk 5 above)
Comment (VNTee) [reply src/main/java/com/epam/validator/Validator.java:54]: Done

**Review by MaltsevaKsenia — APPROVED** [review]

**Review by ghost — APPROVED** [review]

**Review by goponenkoev — APPROVED** [review]

**Pull request merged**